### PR TITLE
[SPARK-49097][INFRA] Add Python3 environment detection for the `build_error_docs` method in `build_api_decs.rb`

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR aims to add Python3 environment detection for the `build_orror_docs` method in `build_api_decs.rb`.

### Why are the changes needed?

Make the environment exception prompts more friendly for developers when generating documents.
Before:
image
After:
image

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: docs/_plugins/build_api_docs.rb [hunk 1]
```diff
@@ -194,7 +194,12 @@ def build_sql_docs
 
 def build_error_docs
   print_header "Building error docs."
-  system("python '#{SPARK_PROJECT_ROOT}/docs/util/build-error-docs.py'") \
+
+  if !system("which python3 >/dev/null 2>&1")
+    raise("Missing python3 in your path, stopping error doc generation")
+  end
+
+  system("python3 '#{SPARK_PROJECT_ROOT}/docs/util/build-error-docs.py'") \
```
Comment (wayneguow) [inline docs/_plugins/build_api_docs.rb:202]: Since users may not use python venv, I suggest changing to `python3` here.

**Review by dongjoon-hyun — COMMENTED** [review]
It seems the failure is relevant to this change.
Could you make CI happy, @wayneguow ?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -872,6 +872,8 @@ jobs:
       run: |
         # We need this link because the jekyll build calls `python`.
         ln -s "$(which python3.9)" "/usr/local/bin/python"
+        # We need this link to make sure `python3` points to `python3.9` which contains the prerequisite packages.
+        ln -s "$(which python3.9)" "/usr/local/bin/python3"
```
Comment (wayneguow) [inline .github/workflows/build_and_test.yml:876]: Currently, there are multiple pythons  in the ci env： 
- Python 3.9.19
- Python 3.10.12
 And python3 points to `Python 3.10.12`.

**Comment by wayneguow** [comment]
> It seems the failure is relevant to this change. Could you make CI happy, @wayneguow ?

Well, I found out the reason, because python3 currently does not point to the python3.9 with the installation packages we need.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -872,6 +872,8 @@ jobs:
       run: |
         # We need this link because the jekyll build calls `python`.
         ln -s "$(which python3.9)" "/usr/local/bin/python"
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:874]: Can we remove the above two line from [SPARK-46935](https://issues.apache.org/jira/browse/SPARK-46935), @wayneguow ?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test.yml (hunk 3 above)
Comment (wayneguow) [reply .github/workflows/build_and_test.yml]: Yes, good advice, this link is indeed introduced in this PR #44971 and only used in this place.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @wayneguow and @HyukjinKwon .
Merged to master.

**Pull request closed without merging**